runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -725,7 +726,7 @@ impl<T: Config> Pallet<T> {
 							// That way we avoid possible duplicate checks while assuring all
 							// backed candidates fine to pass on.
 							check_ctx
-								.verify_backed_candidate(parent_hash, candidate_idx, backed_candidate)
+								.verify_backed_candidate(parent_hash, parent_storage_root, candidate_idx, backed_candidate)
```

Comment:
I don't think this is correct, we'd want to filter out all of them, not only the outer error

Reply:
I'm having a hard time connecting your comment to the code here. Can you explain in more detail?

Reply:
We are now only considering the outer error of the nested result, I'd argue that an error of the inner result type is also a good enough reason to discard the backed candidate iiuc. This can be deferred to a follow up PR, since as you correctly stated, behavior does not change compared to current master

Reply:
> I'd argue that an error of the inner result type is also a good enough reason to discard the backed candidate iiuc

Ok, fair. We should probably discard all backed candidates when encountering this error. It means something is seriously wrong and governance will probably need to step in.

Reply:
I assume we can change that in the next release.